pkg/api/schema/database_protoconv.go

```diff
@@ -61,14 +61,17 @@ func TxFromProto(stx *Tx) *store.Tx {
 
 	tx := store.NewTxWithEntries(entries)
 
-	tx.ID = stx.Header.Id
-	tx.PrevAlh = DigestFromProto(stx.Header.PrevAlh)
-	tx.Ts = stx.Header.Ts
-	tx.BlTxID = stx.Header.BlTxId
-	tx.BlRoot = DigestFromProto(stx.Header.BlRoot)
+	hdr := tx.Header()
+
+	hdr.ID = stx.Header.Id
+	hdr.Ts = stx.Header.Ts
+	hdr.PrevAlh = DigestFromProto(stx.Header.PrevAlh)
+	hdr.BlTxID = stx.Header.BlTxId
+	hdr.BlRoot = DigestFromProto(stx.Header.BlRoot)
+	hdr.Version = int(stx.Header.Version)
+	hdr.Metadata = TxMetadataFromProto(stx.Header.Metadata)
 
 	tx.BuildHashTree()
-	tx.CalcAlh()
```

Comment:
Why this is removed ?

Reply:
because Alh is directly calculated on the header.
BuildHashTree is setting in the header the root hash of the entries